lib/spear/connection.ex

```diff
@@ -392,9 +392,15 @@ defmodule Spear.Connection do
   end
 
   defp process_response({:done, request_ref}, s) do
```

Comment:
this is set up to handle the Mint `{:done, request_ref}` message returned by [`Mint.HTTP.stream/2`](https://hexdocs.pm/mint/Mint.HTTP.html#stream/2-responses)

see those docs for more info on how that works

basically, on HTTP2 request conclusion:

- pop the request struct out of state
- if it's an open subscription (`from: nil`), send it `{:eos, :dropped}`
- if it's a regular request, e.g. append/4 or such, this is the normal conclusion of the request lifecycle